### PR TITLE
docs: add doc, mlflow on Synapse Analytics

### PR DESCRIPTION
**Pull request by JessicaXYWang**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

_Briefly describe the changes included in this Pull Request._

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change any dependencies?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure the dependencies are resolved correctly, and list changes here.

## Does this PR add a new feature? If so, have you added samples on website?

- [ ] No. You can skip this section.
- [ ] Yes. Make sure you have added samples following below steps.

1. Find the corresponding markdown file for your new feature in `website/docs/documentation` folder.
   Make sure you choose the correct class `estimators/transformers` and namespace.
2. Follow the pattern in markdown file and add another section for your new API, including pyspark, scala (and .NET potentially) samples.
3. Make sure the `DocTable` points to correct API link.
4. Navigate to website folder, and run `yarn run start` to make sure the website renders correctly.
5. Don't forget to add `<!--pytest-codeblocks:cont-->` before each python code blocks to enable auto-tests for python samples.
6. Make sure the `WebsiteSamplesTests` job pass in the pipeline.

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/mlflow/mlflow_on_synapse_analytics.md](https://github.com/microsoft/SynapseML/blob/9b87296cc8e1e28b0bcaafb789d197d41cf87e6c/website%2Fdocs%2Fmlflow%2Fmlflow_on_synapse_analytics.md) | 66 | 48 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/4e152fa6-a8ad-46af-93de-143790f3f6cd) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

File: website/docs/mlflow/mlflow_on_synapse_analytics.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+subprocess.check_call([sys.executable, "-m", "pip", "install", "azure.ai.ml"])
+```
+
+### Configure authentication
+
+
+```python
+# Please manage the secrets with Azure KeyVault. The following code example is for quick test not for production.
+import os
+os.environ["AZURE_TENANT_ID"] = "<tenantId> in the service principle json above"
```
Comment (mhamilton723) [inline website/docs/mlflow/mlflow_on_synapse_analytics.md:61]: COnsider making this a real notebook as opposed to an untested markdown. You can use the find_secret api to automate the keyvault integration

File: website/docs/mlflow/mlflow_on_synapse_analytics.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+model = LightGBMClassifier(
+    objective="binary", featuresCol="features", labelCol="Bankrupt?", isUnbalance=True
+)
+from synapse.ml.lightgbm import LightGBMClassifier
+
+with mlflow.start_run():
+    fit_model = model.fit(train_data)
+    mlflow.spark.log_model(
+        fit_model, artifact_path=artifact_path, registered_model_name=model_name
+    )
```
Comment (mhamilton723) [inline website/docs/mlflow/mlflow_on_synapse_analytics.md:145]: please combine the vector assembler and LightGBM into a single pipeline and log the whole thing

File: website/docs/mlflow/mlflow_on_synapse_analytics.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    .option("header", True)
+    .option("inferSchema", True)
+    .load(
+        "wasbs://publicwasb@mmlspark.blob.core.windows.net/company_bankruptcy_prediction_data.csv"
+    )
+)
+train, test = df.randomSplit([0.85, 0.15], seed=1)
+from pyspark.ml.feature import VectorAssembler
+
+feature_cols = df.columns[1:]
```
Comment (mhamilton723) [inline website/docs/mlflow/mlflow_on_synapse_analytics.md:125]: maybe use something like [c for c in df.columns if c != label_col]

File: website/docs/mlflow/mlflow_on_synapse_analytics.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
+# Log Model with MLflow on Synapse Analytics
```
Comment (mhamilton723) [inline website/docs/mlflow/mlflow_on_synapse_analytics.md:1]: We have a few pages on MLFlow already, so we might be getting to the point where we should factorize and organize them. One potential factorization could be to make this page just cover installation of MLFlow on different environments. The next steps links could then go to the other two documents 

https://microsoft.github.io/SynapseML/docs/mlflow/examples/
https://microsoft.github.io/SynapseML/docs/mlflow/autologging/

Those documents should contain pointers back to this document in the prerequisites. Likewise, the overview doc

https://microsoft.github.io/SynapseML/docs/mlflow/introduction/

Should point to this installation doc to guide users through. Also one thing that might make this simpler too is to combine the demo in this doc, with the other two markdowns on example usage and autologging and make just a single usage document that gives people a few minimal examples, and goes over how to configure autologging

**Pull request closed without merging**